public/app/plugins/datasource/loki/datasource.test.ts

```diff
@@ -273,10 +273,13 @@ describe('LokiDatasource', () => {
     });
 
     it('should interpolate before adding adhoc filters', async () => {
+      const originalQuery = 'rate({bar="baz", job="foo"} |= "bar" [$__auto])';
```

Comment:
Can we try `avg_over_time({bar="baz", job="foo"} |= "bar" |unwrap foo[$__auto]) by (foo,bar,buzz)`

Reply:
For this test and these changes it doesn't make any difference, specially because we're mocking `templateSrv`.

I can tell you that, with this branch, Explore Logs produces:

`avg_over_time(${filters} ${patterns} ${logsFormat} ${fields} ${lineFilter} | unwrap  caller [$__auto]) by (detected_level,eventType,component,level,msg)`

And `modifyQuery` receives:

`avg_over_time({service_name=`distributor`}   | logfmt   | unwrap  caller [__V_0____auto__V__]) by (detected_level,eventType,component,level,msg)`

![imagen](https://github.com/grafana/grafana/assets/1069378/791119b4-2063-4a67-a19b-24dc6d450e6d)